cmd/provider-services/cmd/run.go

```diff
@@ -746,7 +746,9 @@ func doRunCmd(ctx context.Context, cmd *cobra.Command, _ []string) error {
 		return err
 	}
 
-	err = gwgrpc.NewServer(ctx, grpcaddr, []tls.Certificate{tlsCert}, service)
+	ctx = gwgrpc.ContextWithQueryClient(ctx, cl.Query())
+
+	err = gwgrpc.Serve(ctx, grpcaddr, []tls.Certificate{tlsCert}, service)
```

Comment:
I see were you going with it, however, this approach has a few caveats:
- there is error check prior serve starts.
- calling `Serve` makes assumptions everything is setup up, which is not true in this case, see above

I think better approach would be to `NewServer` as is, or

remove both `group.Go` from `NewServer` and put them into `Serve`. that implies some extra check as well as `Serve` to be blocking and error check by caller

Reply:
I'm not 100% sure what you mean here. `Serve` currently has all the logic for running the server. The only change that affects `run.go` is that the function name changed. The implementation should be the same with a sub function (`newServer`) that returns the gRPC server to make testing easier.

In other words, I think I've already implemented it in the way you are requesting.